### PR TITLE
Keybind for Extend/shrink selection up and down

### PR DESCRIPTION
**Pull request by EmiOnGit**
Implements  #9079

### CODE REVIEW

File: helix-term/src/commands.rs
Comment (chtenb) [inline helix-term/src/commands.rs:2442]: I noticed that `select_line_up` does not work when started on an empty line.

File: helix-term/src/commands.rs
Comment (chtenb) [inline helix-term/src/commands.rs:2440]: The case where you flip over the selection (e.g. JJKKK) feels a little off. I can think of two ways of making this more graceful:

IF selection does not consist of whole lines THEN expand selection to whole lines
ELSE IF selection consists of a single line and orientation is not in the direction of movement THEN change selection orientation
ELSE modify selection like you already implemented

Or alternatively, adopt the behavior from the bindings at https://github.com/helix-editor/helix/issues/165#issuecomment-1817779737

File: helix-term/src/commands.rs
Comment (EmiOnGit) [reply helix-term/src/commands.rs]: I agree, In fact I don't know why I didn't call it above/below especially since I used these worlds in the description to begin with. Renamed them now

**Comment by EmiOnGit** [comment]
The implementation should work better now. 
Especially when flipping the selection and starting on a empty line :)
I am unsure about the behaviour of flipping the selection as soon as the selection is one line.
e.g. JJK currently leaves us with the cursor on the line start instead of the end.
How do you feel about that? I'm open to change that.

File: helix-term/src/commands.rs
Comment (EmiOnGit) [reply helix-term/src/commands.rs]: The new version uses the behavior from your bindings mostly!

**Review by Vulpesx — APPROVED** [review]
this appears fine as is, there are a lot of pull requests though so it will be a while before 2 people with write access approve it.

**Comment by EmiOnGit** [comment]
I don't mind as I can just use my personal branch for now. 
Hope it will get merged eventually as I see people asking about this behaviour frequently.

**Comment by EmiOnGit** [comment]
Wouldn't say that the lint improves readability of the code but it's fine either way I guess. Sorry for not checking beforehand

**Review by the-mikedavis — CHANGES_REQUESTED** [review]

File: helix-term/src/commands.rs
Comment (EmiOnGit) [reply helix-term/src/commands.rs]: Thought copying a bool is not only more readable, but also faster than referencing. Doesn't matter in the end so I reverted it

File: helix-term/src/commands.rs
Comment (EmiOnGit) [reply helix-term/src/commands.rs]: That's a great point!

File: helix-term/src/commands.rs
Comment (EmiOnGit) [reply helix-term/src/commands.rs:2418]: True, I rearranged the functions :)

**Review by the-mikedavis — APPROVED** [review]

**Review by pascalkuthe — APPROVED** [review]

**Comment by chtenb** [comment]
Nice that this made it in. In case people are looking for ways to bind this, the following is a clean way to integrate this in the default keymapping.

```toml
[keys.normal]
J = ["select_line_below"]
K = ["select_line_above"]

"A-j" = "join_selections"
"A-J" = "join_selections_space"

X = "remove_primary_selection"
x = "keep_selections"
"A-x" = "remove_selections"

[keys.select]
J = ["select_line_below"]
K = ["select_line_above"]

"A-j" = "join_selections"
"A-J" = "join_selections_space"

X = "remove_primary_selection"
x = "keep_selections"
"A-x" = "remove_selections"
```

The bonus is that `X` pairs ergonomically with `C` and `(`/`)`.

**Pull request merged**
